evm-ds/src/convert.rs

```diff
@@ -0,0 +1,180 @@
+use crate::protos::Evm as EvmProto;
+use byteorder::{BigEndian, ByteOrder};
+use bytes::Bytes;
+use primitive_types::*;
+use std::ops::Shr;
+
+impl From<H160> for EvmProto::Address {
+    fn from(address: H160) -> Self {
+        let address = address.as_fixed_bytes();
+        let mut addr = EvmProto::Address::new();
+        addr.set_x0(BigEndian::read_u32(&address[0..4]));
+        addr.set_x1(BigEndian::read_u64(&address[4..12]));
+        addr.set_x2(BigEndian::read_u64(&address[12..20]));
+        addr
+    }
+}
+
+impl From<&EvmProto::Address> for H160 {
+    fn from(address: &EvmProto::Address) -> Self {
+        let mut buf = [0u8; 20];
+        BigEndian::write_u32(&mut buf[0..4], address.x0);
+        BigEndian::write_u64(&mut buf[4..12], address.x1);
+        BigEndian::write_u64(&mut buf[12..20], address.x2);
+        H160::from_slice(&buf)
+    }
+}
+
+impl From<U256> for EvmProto::UInt256 {
+    fn from(num: U256) -> Self {
+        let mut result = Self::new();
+        result.set_x3(num.low_u64());
+        result.set_x2(num.shr(64).low_u64());
+        result.set_x1(num.shr(128).low_u64());
+        result.set_x0(num.shr(192).low_u64());
```

Comment:
Would this be clearer?
```suggestion
        // `U256` stores bytes in little-endian order.
        let U256([x3, x2, x1, x0]) = num;
        result.set_x3(x3);
        result.set_x2(x2);
        result.set_x1(x1);
        result.set_x0(x0);
```

Reply:
Is little endian order of U256 guaranteed though?

Reply:
Fair point - I guess I'd argue yes because it says so in the documentation, but maybe you're right that explicitly asking for the low word is better.

> Little-endian large integer type 256-bit unsigned integer.